lightning-persister/src/fs_store.rs

```diff
@@ -215,21 +241,36 @@ impl FilesystemStoreInner {
 		}
 
 		let res = {
-			let inner_lock_ref = {
+			let inner_lock_ref;
+			let mut async_state;
+			{
 				let mut outer_lock = self.locks.lock().unwrap();
-				Arc::clone(&outer_lock.entry(dest_file_path.clone()).or_default())
-			};
-			let mut last_written_version = inner_lock_ref.write().unwrap();
+				inner_lock_ref = Arc::clone(&outer_lock.entry(dest_file_path.clone()).or_default());
```

Comment:
I really really tried to extract this block (which is duplicated for `remove`) into a function, but I couldn't get it to work. Returning the guard that references `inner_lock_ref` was the problem all the time. Returning both as a tuple or struct didn't fix it either.